### DISCUSSION TITLE
Pass custom values to zwave device

### DISCUSSION THREAD

**Discussion by dersgniw** [zwave-js/zwave-js-ui#2270] [Q&A]
My zooz zen24 switch seems to be missing a feature.  Specifically, when turning on/off via home assistant, the ramp rate is ignored.  I have the correct settings I believe.

Can I mess around with the commands that are sent to the switch?  I'd like to do some troubleshooting, but I don't know where to start really. I've been staring at debug logs so I want to try passing different values for duration and maybe for setting 17.  

Any ideas?

**Comment by robertsLando** [comment]
@dersgniw You could use driver function for this. Just write your custom function you want to call using plain js code :)

**Comment by dersgniw** [comment]
Are there any examples of this?  

One wrinkle to this solution might be that I'm running in a docker container.

**Reply by robertsLando** [reply]
@dersgniw if you want to call it from outside you have to check mqtt apis docs, otherwise you can call it directly from control panel ui under advanced > driver function

**Reply by dersgniw** [reply]
Ok, time to break stuff.  Thanks!

**Reply by dersgniw** [reply]
A couple tries here.  I'm sure you have more to do than tell me what I'm doing wrong, but I'll ask anyway.

```
const {MultilevelSwitchCCSet} = this.require('zwave-js');
const cc = new MultilevelSwitchCCSet(driver, {nodeId: 9 ,targetValue: 55, duration:"10s" ,}); 
await driver.sendCommand(cc);
```
And
```
const {MultilevelSwitchCCSet, SupervisionCCGet, SupervisionCC } = this.require('zwave-js');
const cc = new MultilevelSwitchCCSet(driver, {nodeId: 9 ,targetValue: 55, duration:"10s" ,}); 
const cc2 = SupervisionCC.encapsulate(driver, cc);
await driver.sendCommand(cc2);
```

The problem is this only works when I remove the duration parameter.

I'm getting this error:

2022-02-18T17:29:18.867Z DRIVER   Unexpected error during SerialAPI command: TypeError: this.duration.serializeS
                                  et is not a function

**Reply by robertsLando** [reply]
Wait a moment, did you also tried from control panel UI to set duration in set value option when sending a value to targetValue prop? 

Btw you could use zwavejs setValue api and pass options as second argument to do this

**Reply by brianpettit** [reply]
That part of the interface is a little confusing actually.

Setting [9-38-0-duration] Transition duration to any value gives me this error - Driver: Multilevel Switch: "duration" is not a supported property (ZW0322) - (The error is in red at the top of the page even after page reload.  I have to turn off and back on logging to get rid of it).

The "[7-38-0-Up] Perform a level change (Up) On/Off" are also a little confusing.  They seem to send a command that can be stopped, but upon refreshing the page they remain toggled.

Interestingly, for my zen74, the duration is sent to the switch when I use the send options config parameter, but it seems to be ignored and it uses whatever the configured ramp rate is, even if it's set to zero.  The configured ramp rate is used when clicking in Home Assistant as well.

The zen24 ignores the configured ramp rate when I use home assistant, but it actually uses the duration in the send options if I pass it.  There is a  Z-Wave Ramp Rate Control setting which can be "match" or "independent", but that doesn't seem to make a difference here.

Thanks for listening.  I know very little here, but it's really interesting stuff.

**Reply by brianpettit** [reply]
Sorry, that's my work account.  Same person here.

**Reply by kpine** [reply]
In the control panel you use the transition **option**. Duration is not a settable property. Also ensure the Multilevel Switch CC is V2+. V1 does not support duration (transition).

![image](https://user-images.githubusercontent.com/791794/154831993-6b0ec7bd-0053-4bb4-9a54-4b1f498e4c62.png)

**Comment by kpine** [comment]
In HA, use the `transition` service data attribute with the [light.turn_on](https://www.home-assistant.io/integrations/light/#service-lightturn_on) service.

```yaml
service: light.turn_on
data:
  transition: 10
target:
  entity_id: light.dining_room
```

**Reply by brianpettit** [reply]
Kpine, how do you call that from a "light card" in the HA UI?

**Reply by kpine** [reply]
I don't think that's possible. Maybe you can find another card to suit your needs.

**Reply by dersgniw** [reply]
I did find this light_profiles.csv file and I can add default transition values in there.

However, the transition when using Home Assistant is really jerky.  Using the switch the transition to on/off is smooth.

**Reply by kpine** [reply]
> I did find this light_profiles.csv file and I can add default transition values in there.

Yes, light profiles work if setting a default transition fits your use case.

> However, the transition when using Home Assistant is really jerky. Using the switch the transition to on/off is smooth.

That's an issue with the device then.
